pkg/operator/ceph/cluster/osd/spec.go

```diff
@@ -567,31 +639,148 @@ func (c *Cluster) provisionPodTemplateSpec(osdProps osdProperties, restart v1.Re
+// otherwise we will end up with a new conflict during the job/deployment initialization
+func (c *Cluster) getPVCMetadataInitContainer(mountPath string, osdProps osdProperties) v1.Container {
+	return v1.Container{
+		Name:  blockPVCMetadataMapperInitContainer,
+		Image: c.cephVersion.Image,
+		Command: []string{
+			"cp",
+		},
+		Args: []string{"-a", fmt.Sprintf("/%s", osdProps.metadataPVC.ClaimName), fmt.Sprintf("/srv/%s", osdProps.metadataPVC.ClaimName)},
+		VolumeDevices: []v1.VolumeDevice{
+			{
+				Name:       osdProps.metadataPVC.ClaimName,
+				DevicePath: fmt.Sprintf("/%s", osdProps.metadataPVC.ClaimName),
+			},
+		},
+		VolumeMounts: []v1.VolumeMount{
+			{
+				MountPath: "/srv",
+				Name:      fmt.Sprintf("%s-bridge", osdProps.metadataPVC.ClaimName),
+			},
+		},
+		SecurityContext: opmon.PodSecurityContext(),
+		Resources:       osdProps.resources,
+	}
+}
+
+func (c *Cluster) getPVCMetadataInitContainerActivate(mountPath string, osdProps osdProperties) v1.Container {
+	return v1.Container{
+		Name:  blockPVCMetadataMapperInitContainer,
+		Image: c.cephVersion.Image,
+		Command: []string{
+			"cp",
+		},
+		Args: []string{"-a", fmt.Sprintf("/%s", osdProps.metadataPVC.ClaimName), path.Join(mountPath, "block.db")},
+		VolumeDevices: []v1.VolumeDevice{
+			{
+				Name:       osdProps.metadataPVC.ClaimName,
+				DevicePath: fmt.Sprintf("/%s", osdProps.metadataPVC.ClaimName),
+			},
+		},
+		// We need to call getPvcOSDBridgeMountActivate() so that we can copy the metadata block into the "main" empty dir
+		// This empty dir is passed along every init container
+		VolumeMounts:    []v1.VolumeMount{getPvcOSDBridgeMountActivate(mountPath, osdProps.pvc.ClaimName)},
+		SecurityContext: opmon.PodSecurityContext(),
+		Resources:       osdProps.resources,
+	}
+}
+
+func (c *Cluster) getActivatePVCInitContainer(osdProps osdProperties, osdID string) v1.Container {
+	osdDataPath := activateOSDMountPath + osdID
+	osdDataBlockPath := path.Join(osdDataPath, "block")
+
+	container := v1.Container{
+		Name:  activatePVCOSDInitContainer,
+		Image: c.cephVersion.Image,
+		Command: []string{
+			"ceph-bluestore-tool",
+		},
+		Args:            []string{"prime-osd-dir", "--dev", osdDataBlockPath, "--path", osdDataPath, "--no-mon-config"},
+		VolumeMounts:    []v1.VolumeMount{getPvcOSDBridgeMountActivate(osdDataPath, osdProps.pvc.ClaimName)},
+		SecurityContext: opmon.PodSecurityContext(),
+		Resources:       osdProps.resources,
+	}
+
+	return container
+}
+
+func (c *Cluster) getExpandPVCInitContainer(osdProps osdProperties, osdID string) v1.Container {
+	osdDataPath := activateOSDMountPath + osdID
+
+	container := v1.Container{
+		Name:  expandPVCOSDInitContainer,
+		Image: c.cephVersion.Image,
+		Command: []string{
+			"ceph-bluestore-tool",
+		},
+		Args:            []string{"bluefs-bdev-expand", "--path", osdDataPath},
```

Comment:
Is this command always available on nautilus or newer?

Reply:
Yes, available since Luminous AFAIR.